trace/kafka/kafka.go

```diff
@@ -7,21 +7,26 @@ import (
 	"github.com/beatlabs/patron/encoding/json"
 	"github.com/beatlabs/patron/errors"
 	"github.com/beatlabs/patron/trace"
-	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
 )
 
 // Message abstraction of a Kafka message.
 type Message struct {
 	topic string
 	body  []byte
+	key   *string
 }
 
 // NewMessage creates a new message.
 func NewMessage(t string, b []byte) *Message {
 	return &Message{topic: t, body: b}
 }
 
+func NewMessageWithKey(t string, b []byte, k *string) *Message {
```

Comment:
Since you want a message with key, key cannot be nil. We should then check if `key == ""` and return an error.
```suggestion
func NewMessageWithKey(t string, b []byte, k string) *Message {
```

Reply:
Sorry for not seeing this earlier...